### DISCUSSION TITLE
[@types/markdown-it]: incorrect quotes option type

### DISCUSSION THREAD

**Discussion by shellscape** [DefinitelyTyped/DefinitelyTyped#55406] [Issues with a @types package] [answered]
_(cc: @peterblazejewicz @sandersn)_

The type for markdown-it options is currently defined as:

```ts
quotes?: string | string[] | undefined;
```

Which is _almost_ correct. `undefined` is not something that should be assigned to this value (ref: https://github.com/markdown-it/markdown-it/issues/812). Here's the offending line: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bc6a410c23072e9a5136cc5ef60a6e82c905a5af/types/markdown-it/lib/index.d.ts#L79

Would love to see this one fixed, as it caused some user confusion.

[a comment by an automated account is not shown]

**Reply by peterblazejewicz** [reply]
@shellscape I understand `undefined` was added to accommodate TS 4.4 changes, to be specific a change to fix long standing problem with detecting missing input form `undefined`:
https://github.com/microsoft/TypeScript/issues/13195
This is not necessarily an error. So every optional property in `Options` has now `undefined` added.
Details of the change:
https://github.com/microsoft/TypeScript/pull/43947

**Reply by shellscape** [reply]
please see the referenced issue; the markdown-it maintainer specifically says that option cannot be undefined, docs reflect that. I understand the intent but it's led to allowing invalid input.

**Reply by peterblazejewicz** [reply]
I'd expect, with a time, a DT definition can fine tune devex when using specific package, to provide best experience when 'strictOptionalProperties' is enabled

**Reply by peterblazejewicz** [reply]
Oh, sorry, I've missed the link. LMC

**Reply by peterblazejewicz** [reply]
this is exactly scenario that 4.4 tries to fix. With current TS one can always provide undefined for optional properties, as optional is deemed equal to undefined. You're correct with `markdown-it`, it uses simple merge solution to provide default values from preset. When one sets `undefined` on the property, it's deemed valid, but the actual code never uses safechecks. For now I've implemented a change to remove this stipulation from this particual property.
At least in the editors, the `undefined` should not be available as valid option to pick, thx!
#55409

**Comment by peterblazejewicz** [comment, marked as answer]
Fixed here: ##55409

**Comment by shellscape** [comment]
Thanks very much for tackling this so quickly
